modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -589,69 +594,71 @@ public void getRetryPolicyExhaustedAndConnectionErrorsAreComputed() throws Excep
     ArrayList<String> debugMessages = new ArrayList<>();
-    Logger oldLogger = setLogger(messageSource, LOGGER_FIELD_NAME, createMockLogger(debugMessages, DEBUG));
+    logger.resetLogs();
+    logger.setLevel(DEBUG);
     messageSource.initialise();
     messageSource.start();
-    verifyLogMessage(debugMessages, "Message source 'source' on flow 'appleFlow' is starting");
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    verifyLogMessage(logger.getMessages(), "Message source 'source' on flow 'appleFlow' is starting");
+    logger.resetLevel();
   }
 
   @Test
   public void sourceStoppedLogMessage() throws Exception {
-    ArrayList<String> debugMessages = new ArrayList<>();
-    Logger oldLogger = setLogger(messageSource, LOGGER_FIELD_NAME, createMockLogger(debugMessages, DEBUG));
+    logger.resetLogs();
+    logger.setLevel(DEBUG);
     messageSource.initialise();
     messageSource.start();
     messageSource.stop();
-    verifyLogMessage(debugMessages, "Message source 'source' on flow 'appleFlow' is stopping");
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    verifyLogMessage(logger.getMessages(), "Message source 'source' on flow 'appleFlow' is stopping");
+    logger.resetLevel();
   }
 
 
   @Test
   public void getRetryPolicyExhaustedAndLogShutdownMessage() throws Exception {
     ArrayList<String> errorMessages = new ArrayList<>();
-    Logger oldLogger = setLogger(messageSource, LOGGER_FIELD_NAME, createMockLogger(errorMessages, ERROR));
+    logger.resetLogs();
+    logger.setLevel(ERROR);
     start();
     ConnectionException e = new ConnectionException(ERROR_MESSAGE);
     doThrow(e).when(source).onStart(any());
     messageSource.onException(e);
     new PollingProber(TEST_TIMEOUT, TEST_POLL_DELAY).check(new JUnitLambdaProbe(() -> {
-      verifyLogRegex(errorMessages,
+      verifyLogRegex(logger.getMessages(),
                      "Message source 'source' on flow 'appleFlow' could not be reconnected. Will be shutdown. (.*)");
       return true;
     }));
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    logger.resetLevel();
   }
 
   @Test
   public void reconnectAndLogSuccessMessage() throws Exception {
     start();
-    ArrayList<String> infoMessages = new ArrayList<>();
-    Logger oldLogger = setLogger(messageSource, LOGGER_FIELD_NAME, createMockLogger(infoMessages, INFO));
+    logger.resetLogs();
+    logger.setLevel(INFO);
     ConnectionException e = new ConnectionException(ERROR_MESSAGE);
     messageSource.onException(e);
     new PollingProber(TEST_TIMEOUT, TEST_POLL_DELAY).check(new JUnitLambdaProbe(() -> {
-      verifyLogMessage(infoMessages, "Message source 'source' on flow 'appleFlow' successfully reconnected");
+      verifyLogMessage(logger.getMessages(), "Message source 'source' on flow 'appleFlow' successfully reconnected");
       return true;
     }));
-    setLogger(messageSource, LOGGER_FIELD_NAME, oldLogger);
+    logger.resetLevel();
```

Comment:
put this in an after of the test so that it will always be invoked even if an exception happens.